### PR TITLE
Relion2.1 support

### PR DESCRIPTION
**Pull request by azazellochg**
This is a rather big PR adding Relion 2.1 support.

- added relion2.1 new metadata labels
- added new setRelionAttributes function (to be used in future)
- now rlnGroupName is propagated after ref/classifications (#1121, #1126)
- rlnMicrographName is now set to micName if available otherwise it is set to micId (#1145)
- add relion2 local res protocol(#694)
- add relion2.1 initial model protocol
- added new relion2 options in several protocols
- added CTF options in particles subtraction protocol
- movie refinement now produces output (!) (movie particles)
- polishing protocol completely refactored, now accepts movie particle set as input
- bugfixes (#1014, #564, #1152, summaries added, many other fixes)

Remaining issues:
- [ ] Inf values in star files (#1110)
- [ ] Changes (especially movie ref & polishing) were not tested in relion 1.3
- [ ] Maybe we need to create some kind of viewer for relion local res protocol

So far I have tested the whole relion workflow in versions 2.1 and 1.4 (betagal tutorial data) and it seems to work. I do not have relion1.3 here but will try to install and test as well. 

Please feel free to comment, discuss or participate in testing.

### CODE REVIEW

**Comment by delarosatrevin** [comment]
Thank you so much @azazellochg!!! This is a very important update, users will appreciate it a lot! My suggestion will be that we can forget about R1.3 and move on. I think as packages versions raise, we will need to necessarily remove support for some of them. 

We will join for the testing and provide feedback on this.

**Comment by josuegbl** [comment]
Hi @azazellochg, very very good job doing this update. Thank you so much. Im testing the protocols right now, but I dont see any test for the new protocols: 3d initial model and local resolution. Could you add the test? If you dont, Ill try to add next week.

**Comment by azazellochg** [comment]
Hi @josuegbl , I totally forgot about the tests... I'll try to add them tonight.

**Comment by jelka71** [comment]
Hi @azazellochg, I been looking very much forward for this update. Although when I merge it with release-1.1.1, I get this error on trying to open my existing project: "invalid literal for int() with base 10: '1.0' "
It looks okay when opening new projects.

**Comment by azazellochg** [comment]
Hi @jelka71 ,
Sorry about that, I've just pushed a small change, so this error should go away if you pull the branch.

**Comment by delarosatrevin** [comment]
@azazellochg There are some conflicts here? 
@jelka71 Did you tested after @azazellochg bug fix? Any other issue. 

@azazellochg, I will test more this PR after Monday that I will be back to Stockholm...anything that you want me to take a look (apart from the inf values ;) ?

**Comment by azazellochg** [comment]
@delarosatrevin Conflicts arose after f2fafc6. I didn't test movie particle extraction after this commit.  
Another left to consider: test with different frame range [(here)](https://github.com/I2PC/scipion/pull/1157/files#diff-eb09fc6cfd4d30ca109bf8e6d585503eR1122). I don't know why there was +1 before..

**Comment by delarosatrevin** [comment]
OK, I will try to merge it and @josuegbl can test if his changes are still working in this branch.

File: pyworkflow/em/packages/relion/protocol_base.py [hunk 1]
```diff
@@ -529,7 +529,7 @@ def _defineParams(self, form):
                                pointerClass='SetOfMovieParticles',
                                allowsNull=True, important=True,
                                label='Input movie particles')
-                group.addParam('movieAvgWindow', IntParam, default=5,
+                group.addParam('movieAvgWindow', FloatParam, default=5,
```
Comment (azazellochg) [inline pyworkflow/em/packages/relion/protocol_base.py]: thanks!

**Pull request merged**